Adjust react-select style

vlaux opened the pull request:
#### What is the purpose of this pull request?
Add some custom styles to react-select in order to correspond to our design system style.

#### What problem is this solving?
Since height, padding and font size from select were different from other inputs, they use to get unaligned when together:
![image](https://user-images.githubusercontent.com/5256673/84812643-143af080-afe5-11ea-8c2a-6282e60b61a7.png)
Placeholder font size:
![image](https://user-images.githubusercontent.com/5256673/84812787-519f7e00-afe5-11ea-8198-9a1427a3010c.png)


#### How should this be manually tested?
[Running workspace](https://vlauxbeta--pricingqa.myvtex.com/admin/promotions/1a18dd71-26ae-428b-a425-af6d816af784/)

#### Screenshots or example usage

![image](https://user-images.githubusercontent.com/5256673/84812717-359bdc80-afe5-11ea-9f59-a5419884e9f6.png)

![image](https://user-images.githubusercontent.com/5256673/84812737-3fbddb00-afe5-11ea-9090-b626a2192faa.png)

#### Types of changes

- [X] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

vlaux commented:
There may be an issue with font size for different select's `size` variants.
Since `regular` is the default, should we only apply the input's default font size to this version, while `small` and `large` use other font sizes? cc @jonathasbsouza

vlaux commented:
After talking to @jonathasbsouza, I decided to convert this to draft until we fix all size variants (small, regular, large) to correspond our updated input's style

davicosta99 commented:
We discussed the component at the working session. Here's a summary:
- Tag vertical alignment. 16pt font-size in the large variant. (Maybe reuse the tag component)
- Focus state and error state not aligned with the Input component. Focus state shouldn't have a hover behavior.
- Ideally, when a tag was at the beginning, the field's left padding would be 8px, instead of 16px.

vlaux commented:
So, I did some adjustments:
* Fixed hover & focus border color
* Fixed error state
* Using 16px font size on tags for `large` variant
* Reduced left padding on multi input selects from 16px to 4px (as discussed with @davicosta99, 8px was not enough)

I couldn't really solve the vertical alignment in a consistent way, so I let it as it is for now. If someone wants to take a look on it, feel free to add it!
